README.md
[![codecov](https://codecov.io/gh/dwenzel/t3events_reservation/branch/master/graph/badge.svg)](https://codecov.io/gh/dwenzel/t3events_reservation)
[![Build Status](https://travis-ci.org/dwenzel/t3events_reservation.svg?branch=master)](https://travis-ci.org/dwenzel/t3events_reservation)

t3events_reservation
====================

Reservations is an extension for the TYPO3 CMS. It extends t3events and allows to manage bookings for events.

## Table of contents
- [Installation and requirements](/Documentation/Install.md)
- [Configuration](/Documentation/Configuration.md)
- [Reservation Process](/Documentation/ReservationProcess.md)
- [Routing](/Documentation/Routing.md)
- [Error Handling](/Documentation/ErrorHandling.md)
- [Validation](/Documentation/Validation.md)
- [Notifications](/Documentation/Notifications.md)

## Credits
Thanks for contribution and feedback
* [Sebastian Kreideweiss](https://github.com/kreidewe)

This extension has been developed at [CPS IT GmbH](https://cps-it.de).



ChangeLog
2023-08-09         [RELEASE] Release of t3events_reservation 1.0.0 (Steffen Mevius)$
2023-06-23 212f483 [TASK] change pagination default (Steffen Mevius)
2023-06-19 c531e95 [TASK][43687] Provide setting 'module' for rendering backend module (Oliver Noth)
2023-06-19 9b1ba4b [TASK][43687] Fix rendering backend module (Oliver Noth)
2023-06-14 f6006c6 [TASK] add pagination (Steffen Mevius)
2023-06-07 a5e5525 [TASK] add missing abstract functions (Steffen Mevius)
2023-05-31 6181827 [TASK] fix composer constraints (Steffen Mevius)
2023-05-31 7e1249a [TASK] fix composer.json constraint (Steffen Mevius)
2023-05-31 f807dc9 [TASK] run rector with php 81 ruleset (Steffen Mevius)
2023-05-31 fe680aa [TASK] make compoer.json php 8.1 compatible (Steffen Mevius)
2021-08-19 cc7dcaa [TASK] Adjust code for booking process (Santiago Nunez Negrillo)
2021-08-12 c876e47 [TASK] Register extension icons (Santiago Nunez Negrillo)
2021-08-12 046a85e [TASK] Rewrite extbase commands into TYPO3 v10 commands (Santiago Nunez Negrillo)
2021-08-09 ac7af07 [TASK] Add extension key to ReservationController to fix errors in translations (Santiago Nunez Negrillo)
2021-08-09 44ff27b [TASK] Add reservation argument to request if it is set in contact argument (Santiago Nunez Negrillo)
2021-08-09 eeaba53 [TASK] Adjust session injection for ReservationAccessTrait (Santiago Nunez Negrillo)
2021-08-05 e0edec5 [TASK] Adjust templates to transform deprecated uploads fields to FAL fields (Santiago Nunez Negrillo)
2021-08-04 4008811 [TASK] Use EditUriViewHelper instead of EditRecordViewHelper in Bookings List view (Santiago Nunez Negrillo)
2021-08-04 0c2b179 [TASK] Adjust TCA configuration for TYPO3 v10 (Santiago Nunez Negrillo)
2021-08-04 d30d352 [TASK] Adjust type for variable lesson in Reservation because Extbase cannot handle two types (Santiago Nunez Negrillo)
2021-08-04 633fa87 [TASK] Change Extbase objects and persistence configuration to TYPO3 v10 syntax (Santiago Nunez Negrillo)
2021-07-21 a7c947d [TASK] Set version of dwenzel/t3events to dev-feature/version10compatible (Santiago Nunez Negrillo)
2021-07-21 2ba643c [TASK] Change deprecated "getValidationResults" function with new one "validate" (Santiago Nunez Negrillo)
2021-07-21 de737e1 [TASK] Change old PHP annotations with new ones (Santiago Nunez Negrillo)
2021-07-21 a9d7aad [TASK] Adjust plugin and module configuration for TYPO3 v10 (Santiago Nunez Negrillo)
2021-07-21 9972efc [TASK] Add Services.yaml configuration file for TYPO3 v10 (Santiago Nunez Negrillo)
2021-07-15 8a4d45c [TASK] Adjust composer.json (Santiago Nunez Negrillo)
2021-07-15 6e60d19 [TASK] Update composer.json and ext_emconf.php for TYPO3 v10 (Santiago Nunez Negrillo)
2019-08-14 e10be13 [CI] support for PHP < 7.2 dropped (Dirk Wenzel)
2019-08-13 cc396de [RELEASE] Release 0.16.0 (Dirk Wenzel)
2019-08-13 f6eddae [BUGFIX] namespace of ExtensionConfiguration call corrected (Dirk Wenzel)
2019-08-13 a2ec728 [TASK] extension configuration refactored - require dwenzel/t3extension_configuration:~1.2 (Dirk Wenzel)
2019-08-13 bf22a83 [TASK] backend module registration refactored - require dwenzel/t3extensiontools:~1.1 - require dwenzel/t3events:~1.1 - require php:^7.2 (Dirk Wenzel)
2019-03-06 1d05700 [RELEASE] Release of t3events_reservation 0.15.1 (Dirk Wenzel) (Dirk Wenzel)
2019-03-06 175548c [BUGFIX] fixed error in unit tests for CleanUpCommandController caused by insufficient mocking (Dirk Wenzel)
2019-03-06 4c564ec [BUGFIX] fixed error in command controller where deletion of records was not persisted (Dirk Wenzel)
2019-02-15 0b1eb83 [RELEASE] Release of t3events_reservation 0.15.0 (Dirk Wenzel) (Dirk Wenzel)
2019-02-15 ccc342d [TASK] use new method equals to compare objects. (Dirk Wenzel)
2018-11-09 644d653 [RELEASE] Release of t3events_reservation 0.14.0 (Dirk Wenzel) (Dirk Wenzel)
2018-10-11 a4a5bd7 [TASK] - new icons (Timo Gockel)
2018-09-07 308f007 [RELEASE] Release of t3events_reservation 0.13.0 (Dirk Wenzel) (Dirk Wenzel)
2018-08-14 237f7b9 [TASK] clarify condition in PersonDemandFactory (Dirk Wenzel)
2018-08-14 f39896b [TASK] use constants instead of strings in PersonDemandFactory (Dirk Wenzel)
2018-08-14 16f5b20 [TASK] task obsolete method createFromSettings removed from ParticipantDemandFactory (already exists in parent) (Dirk Wenzel)
2018-08-14 115b43b [TASK] obsolete css files removed (Dirk Wenzel)
2018-08-14 9d5c19b [BUGFIX] fixed error caused by invalid call of FlashMessageQueue::__call in ReservationAccessTrait Closes #2 (Dirk Wenzel)
2018-07-01 749d778 [TASK] CleanUpCommandController refactored - use constants - method deleteReservationsCommand simplified (Dirk Wenzel)
2018-06-30 719bd36 [TASK] ContactController and ReservationController refactored - use constants - use NotificationServiceTrait (Dirk Wenzel)
2018-06-30 7cc80a2 [TASK] CloseBookingCommandController refactored - use traits - use constants - clarify if conditions - annotations improved (Dirk Wenzel)
2018-06-30 6cdbb66 [TASK] CI: branch configuration for sonar restored (Dirk Wenzel)
2018-06-30 b127d28 [TASK] CI: sonar.projectKey changed to t3events_reservation (Dirk Wenzel)
2018-06-30 3bea1f0 [TASK] CI: obsolete section branches removed from sonarcloud configuration (Dirk Wenzel)
2018-06-30 2445ad7 [TASK] SettingsInterface implemented - use constants instead of strings - annotations improved (Dirk Wenzel)
2018-06-29 7e69f3a [TASK] CI: exclude some folders from sonar analysis (Dirk Wenzel)
2018-06-29 b1c3a46 [TASK] coding style in templates improved (Dirk Wenzel)
2018-06-29 2548461 [TASK] coding style in templates improved (Dirk Wenzel)
2018-06-29 2f660bd [TASK] CI: branches setting added to sonar configuration (Dirk Wenzel)
2018-06-29 df2c77a [TASK] CI: sonar cloud settings moved to file .sonar-project.properties (Dirk Wenzel)
2018-06-29 cde6d35 [TASK] CI: configuration for sonar cloud added. (Dirk Wenzel)
2018-06-29 b1d4327 [TASK] CI failing tests fixed, older TYPO3 versions removed from build matrix (Dirk Wenzel)

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "t3events_reservation".
 *
 * Auto generated 13-03-2018 11:52
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = ['title' => 'Reservations', 'description' => 'Manages reservations for events. Extends t3events ', 'category' => 'plugin', 'author' => 'Dirk Wenzel, Sebastian Kreideweiss', 'author_email' => 'wenzel@cps-it.de, kreideweiss@cps-it.de', 'state' => 'beta', 'uploadfolder' => '0', 'createDirs' => '', 'clearCacheOnLoad' => 0, 'version' => '1.0.0',
	['depends' =>
		['typo3' => '8.7.0-10.99.99', 't3events' => '1.1.0-0.0.0'], 'conflicts' =>
		[], 'suggests' =>
		[]]];

